src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/ElementListWriter.java

```diff
@@ -74,7 +74,9 @@ public static void generate(BaseConfiguration configuration) throws DocFileIOExc
     }
 
     protected void generateElementListFile(DocletEnvironment docEnv) throws DocFileIOException {
-        try (BufferedWriter out = new BufferedWriter(file.openWriter())) {
+        try (Writer fileWriter = file.openWriter();
+             BufferedWriter out = (fileWriter instanceof BufferedWriter b) ? b
+                     : new BufferedWriter(file.openWriter())) {
```

Comment:
Don't call `file.openWriter()` second time, reuse `fileWriter` that you already have.

Reply:
oops. that's what I meant; thanks